coffee-backend/src/main/kotlin/chat/letscoffee/security/security/TokenAuthenticationFilter.kt

```diff
@@ -0,0 +1,49 @@
+package chat.letscoffee.security.security
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
+import org.springframework.util.StringUtils
+import org.springframework.web.filter.OncePerRequestFilter
+import java.io.IOException
+import javax.servlet.FilterChain
+import javax.servlet.ServletException
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+class TokenAuthenticationFilter : OncePerRequestFilter() {
+    @Autowired
+    private val tokenProvider: TokenProvider? = null
```

Comment:
Include these in constructor to get rid of `!!` in the following methods

Reply:
🌤️